helix-term/tests/test/commands.rs

```diff
@@ -385,3 +385,70 @@ async fn test_character_info() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_delete_char_backward() -> anyhow::Result<()> {
+    // don't panic when deleting overlapping ranges
+    test((
+        platform_line("#(x|)# #[x|]#").as_str(),
+        "c<space><backspace><esc>",
+        platform_line("#[\n|]#").as_str(),
+    ))
+    .await?;
+    test((
+        platform_line("#( |)##( |)#a#( |)#axx#[x|]#a").as_str(),
+        "li<backspace><esc>",
+        platform_line("#(a|)##(|a)#xx#[|a]#").as_str(),
```

Comment:
Why is the first range forward while the rest are backward? How does this test pass? Is there a bug here where the order of the ranges isn't being preserved?

Reply:
This is just how range mapping in helix works. We basically delete all chars inside the selection which yields an empty selection. Helix normalizes this to a single char selection but during that normalization the selection direction is lost. For single char selections the direction doesn't really matter tough so it's not usually noticabke in the ui.

This has nothing to do withthese specific commands tough as they dont actually touch the selection at all, it's just how the automatic selection mapping works.

Reply:
Interesting, ok, thanks for the explanation